main/src/library/MutList.flix

```diff
@@ -359,6 +359,37 @@ mod MutList {
             MutList(rc1, ref b @ rc1, ref l @ rc1)
         }
 
+    ///
+    /// Apply `f` to every element in `v` and concatenate the results.
+    ///
+    /// The result is a new mutable list.
+    ///
+    pub def flatMap(rc1: Region[r1], f: a -> MutList[b, r1] \ ef, v: MutList[a, r]): MutList[b, r1] \ { ef, r, r1 } =
+        let MutList(_, ra, rl) = v;
+        let a = deref ra;
+        let l = deref rl;
+        let bss = Array.new(rc1, l);
+        def loop(i, size)  = {
```

Comment:
An alternative way to init the array here, but I'm not sure whether using init with a side-effect is good style or not:
```
    let rsize = ref 0 @ rc1;
    def get(i) = {
        let bs = f(Array.get(i, a));
        rsize := deref rsize + MutList.length(bs);
        bs        
    };
    let bss = Array.init(rc1, get, l);
    let size = deref rsize;
```